distributed/client.py

```diff
@@ -3876,7 +3866,85 @@ def register_worker_callbacks(self, setup=None):
         setup : callable(dask_worker: Worker) -> None
             Function to register and run on all workers
         """
-        return self.sync(self._register_worker_callbacks, setup=setup)
+        return self.register_worker_plugin(_WorkerSetupPlugin(setup))
+
+    @gen.coroutine
+    def _register_worker_plugin(self, plugin=None, name=None):
+        responses = yield self.scheduler.register_worker_plugin(
+            plugin=dumps(plugin), name=name
+        )
+        for response in responses.values():
+            if response["status"] == "error":
+                exc = response["exception"]
+                typ = type(exc)
+                tb = response["traceback"]
+                six.reraise(typ, exc, tb)
+        raise gen.Return(responses)
+
+    def register_worker_plugin(self, plugin=None, name=None):
+        """
+        Registers a lifecycle worker plugin for all current and future workers.
+
```

Comment:
General question: do you have thoughts on using `..versionadded:: <version>` directives? Don't need to do it here necessarily, but I ask since this is being referenced from the dask documentation, which may be on a different release cycle.

Reply:
No strong thoughts.  I've never used them much, either as an author or reader.

My plan was just to merge the doc PR after this gets released.  

Happy with whatever though.

Reply:
Ah, I see that you've since merged the doc PR :)  Shouldn't be a big deal either way